### PR TITLE
Support managing keystore entries during deployment creation and update

### PR DESCRIPTION
**Pull request by dimuon**
## Description

Make use of the modified ECE Deployment API that enables specifying ES keystore contents during Create/Update calls.
It should make it possible to handle ES keystore secrets using deployment resource that got a new [attribute](https://github.com/elastic/terraform-provider-ec/blob/93593c31d81a93330fca4c1f72638376bdc8a57e/docs/resources/deployment.md#nestedatt--elasticsearch--keystore_contents) ([example](https://github.com/elastic/terraform-provider-ec/blob/93593c31d81a93330fca4c1f72638376bdc8a57e/examples/resources/ec_deployment/with-keystore/deployment.tf)).

**The appropriate API change is implemented in ECE 3.7.0, ESS ms-96.
The new IT test (`TestAccDeployment_keystore`) is expected to fail until ESS ms-96 release. I tested it against my local env.**

## Related Issues

Rel: #433

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/ecresource/deploymentresource/read.go [hunk 1]
```diff
@@ -180,6 +181,33 @@ func (r *Resource) read(ctx context.Context, id string, state *deploymentv2.Depl
 		deployment.Elasticsearch.Strategy = baseElasticsearch.Strategy.ValueStringPointer()
 	}
 
+	if baseElasticsearch != nil && deployment.Elasticsearch != nil && !baseElasticsearch.KeystoreContents.IsNull() {
+		ds := baseElasticsearch.KeystoreContents.ElementsAs(ctx, &deployment.Elasticsearch.KeystoreContents, true)
+		diags.Append(ds...)
+
+		// response.Resources.Elasticsearch[0].Info.Metadata.Raw
```
Comment (tobio) [inline ec/ecresource/deploymentresource/read.go:188]:
```suggestion
```

File: ec/ecresource/deploymentresource/elasticsearch/v2/schema.go [hunk 2]
```diff
@@ -150,6 +150,32 @@ func ElasticsearchSchema() schema.Attribute {
 				Optional:    true,
 				Validators:  []validator.String{stringvalidator.OneOf(strategiesList...)},
 			},
+
+			"keystore_contents": keystoreContentsSchema(),
+		},
+	}
+}
+
+func keystoreContentsSchema() schema.Attribute {
```
Comment (tobio) [inline ec/ecresource/deploymentresource/elasticsearch/v2/schema.go:159]: So, we're only planning to manage keystore items which are added to the deployment. That is, we won't remove keystore items added outside of the deployment resource?

Just making sure I'm reading the rest of the code correctly. I agree the above is the right approach.

File: ec/ecresource/deploymentresource/read.go [hunk 3]
```diff
@@ -180,6 +181,33 @@ func (r *Resource) read(ctx context.Context, id string, state *deploymentv2.Depl
 		deployment.Elasticsearch.Strategy = baseElasticsearch.Strategy.ValueStringPointer()
 	}
 
+	if baseElasticsearch != nil && deployment.Elasticsearch != nil && !baseElasticsearch.KeystoreContents.IsNull() {
```
Comment (tobio) [inline ec/ecresource/deploymentresource/read.go:184]: Can we add a comment on this block so it's easier to understand the intent here. IIUC:

* Read TF state into `deployment.Elasticsearch.KeystoreContents`
* Get the current keystore from the API
* Any keystore items in TF state but not in the API response are removed

The above ensures that:
* New keystore items in TF config are added
* Keystore items not managed by TF are left alone
* Keystore items previously managed, but removed from config are removed

Is the above correct?

File: ec/ecresource/deploymentresource/elasticsearch/v2/schema.go (hunk 2 above)
Comment (tobio) [reply ec/ecresource/deploymentresource/elasticsearch/v2/schema.go:159]: Basically just understanding the lack of `computed: true` here as well.

**Review by tobio — APPROVED** [review]
LGTM, some questions to make sure I'm on the same page. I think we can add some comments to the read logic to make this clearer too. 

We need to add a changelog entry here.

File: ec/ecresource/deploymentresource/elasticsearch/v2/schema.go (hunk 2 above)
Comment (dimuon) [reply ec/ecresource/deploymentresource/elasticsearch/v2/schema.go:159]: Yeah, the `ec_deployment` resource should handle only keystore items that are declared in its attribute and shouldn't interfere with the items managed  by `ec_deployment_elasticsearch_keystore` resources. The new IT test should cover this use case.
Also the keystore API is write only so the provider cannot read secrets.

File: ec/ecresource/deploymentresource/read.go (hunk 3 above)
Comment (dimuon) [reply ec/ecresource/deploymentresource/read.go:184]: Thank you for the question - I'll add a comment.

**Pull request merged**